docs/docs/releases/0.4.0.md

```diff
@@ -0,0 +1,209 @@
+    4. **Run the management command** to restrict Django state:
+        ```bash
+        docker exec -it mathesar_service_dev python manage.py add_readonly_django -v 3 mathesar_ro_django
+        ```
+    5. **Update your `docker-compose.yml`** to make Mathesar use the read-only role for Django.
+    6. **Restart the service** to apply the changes.
+
+    The result is an instance that allows login and viewing but blocks any permanent changes to user or app state.
+
+
+*Related work:* [#4571](https://github.com/mathesar-foundation/mathesar/pull/4571 "EXPERIMENTAL: Add read-only Django role creation")
+
+## Documentation
+
+### New public feature roadmap
+
+We’ve [published a roadmap](https://github.com/orgs/mathesar-foundation/projects/2) for Mathesar on our GitHub repository. It outlines the major features and improvements we're planning, so contributors and users can better understand where the project is headed.
+
+We're working on bringing this roadmap to our website as well, with more details and context coming soon. If you’re curious about what's next—or want to get involved—this is a good place to start.
+
+*Related work:*
+[#4586](https://github.com/mathesar-foundation/mathesar/pull/4586 "Create roadmap issue template.")
+[#4630](https://github.com/mathesar-foundation/mathesar/pull/4630 "Roadmap template update")
+[#4631](https://github.com/mathesar-foundation/mathesar/pull/4631 "Add roadmap to README")
+[#4611](https://github.com/mathesar-foundation/mathesar/pull/4611 "Add Mathesar version to bug report")
+[#4614](https://github.com/mathesar-foundation/mathesar/pull/4614 "Updated default type label associated with roadmap item template.")
+
+### New guide for translators
+
+If you’re interested in helping translate Mathesar into your preferred language, we’ve [published a new guide](https://wiki.mathesar.org/engineering/translator-guide/) with everything you need to get started.
+
+### Dataset instructions for developers
+
+We’ve clarified our [instructions for loading and using sample datasets](https://github.com/mathesar-foundation/mathesar/blob/develop/DEVELOPER_GUIDE.md#loading-sample-data) to make it easier for contributors to test features and explore the app with realistic data.
+
+*Related work:* [#4581](https://github.com/mathesar-foundation/mathesar/pull/4581 "Clarify sample datasets in the Developer Guide")
+
+## Maintenance
+
+- Add an application name to all database connections and limit the connection pool size to 2. [#4566](https://github.com/mathesar-foundation/mathesar/pull/4566 "Conn leak bug")
+- Rename Column & Constraint TS types to make way for models [#4562](https://github.com/mathesar-foundation/mathesar/pull/4562 "Rename Column & Constraint TS types to make way for models")
+- Ensure batched RPC responses are ordered as per the order of requests [#4583](https://github.com/mathesar-foundation/mathesar/pull/4583 "Ensure batched RPC responses are ordered as per the order of requests")
```

Comment:
This isn't quite right. The actual change is that the front end now maps RPC function calls to return values using a unique ID (submitted by the front end with each function call object), rather than just the array order when batching. This is to conform with the JSON-RPC 2.0 spec.